### PR TITLE
add tests for recording events

### PR DESCRIPTION
**Pull request by makarandp0**
This change adds integration tests to verify recording events.

update: tests are marked as `skip' pending implementation of JSDK-3064
 
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: test/integration/spec/room.js [hunk 1]
```diff
@@ -42,27 +43,71 @@ describe('Room', function() {
...
   this.timeout(60000);
 
-  (defaults.topology === 'peer-to-peer' ? describe.skip : describe)('.isRecording', () => {
-    [true, false].forEach(enableRecording => {
-      context(`when recording is ${enableRecording ? 'enabled' : 'disabled'}`, () => {
-        let room;
-        let sid;
+  (defaults.topology === 'peer-to-peer' ? describe.skip : describe)('recording', () => {
+    [true, false].forEach(trackSharedAtConnect => {
+      [true, false].forEach(recordingEnabledAtCreate => {
```
Comment (manjeshbhargav) [inline test/integration/spec/room.js:48]: Use `combinationContext()`.

File: test/integration/spec/room.js [hunk 2]
```diff
@@ -42,27 +43,71 @@ describe('Room', function() {
...
-        });
+          if (!trackSharedAtConnect) {
+            // publish track.
+            it(`after track is published ${recordingEnabledAtCreate ? 'fires' : 'does not fire'} recordingStarted`, async () => {
+              const recordingStartedPromise = new Promise(resolve => room.once('recordingStarted', resolve));
+              await room.localParticipant.publishTrack(localTrack);
+
+              if (recordingEnabledAtCreate) {
+                await waitFor(recordingStartedPromise, `recording started: ${room.sid}`);
+                assert.strictEqual(room.isRecording, true);
```
Comment (manjeshbhargav) [inline test/integration/spec/room.js:73]: Please make boolean property verification part of a separate `it()`.

File: test/integration/spec/room.js [hunk 3]
```diff
@@ -42,27 +43,71 @@ describe('Room', function() {
...
+    [true, false].forEach(trackSharedAtConnect => {
+      [true, false].forEach(recordingEnabledAtCreate => {
+        context(`when recording is ${recordingEnabledAtCreate ? 'enabled' : 'disabled'} and tracks ${trackSharedAtConnect ? 'published' : 'not published'} at connect `, () => {
+          let room;
+          let sid;
+          let localTrack;
+          const initialRecording = trackSharedAtConnect ? recordingEnabledAtCreate : false;
+          before(async () => {
+            sid = await createRoom(randomName(), defaults.topology, { RecordParticipantsOnConnect: recordingEnabledAtCreate });
+            localTrack = await waitFor(createLocalVideoTrack(), 'alice local video track');
```
Comment (manjeshbhargav) [inline test/integration/spec/room.js:56]: Please publish 2 tracks and verify `recordingStarted` and `recordingStopped` are fired only once each.

**Review by manjeshbhargav — APPROVED** [review]

**Review by manjeshbhargav — CHANGES_REQUESTED** [review]
Please set the base branch to `develop`.

**Comment by makarandp0** [comment]
@manjeshbhargav I have updated the PR to be against `develop` now.

**Review by manjeshbhargav — APPROVED** [review]

**Pull request merged**